lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -137,22 +137,39 @@ def indent(count, skip1: false)
             result.join('')
           end
 
-          apipie :method, 'Performs a DNS lookup on Foreman server' do
+          apipie :method, 'Performs a Hosts and DNS lookup on Foreman server. Prefers A records over AAAA, on systems without public IPv6 addresses Ruby does not return AAAA record, use the second argument for a specific type.' do
```

Comment:
Not sure how long the whole documentation should be, but it is the Foreman host that performs the lookup. So if you're provisioning a client with IPv6 but Foreman is IPv4-only then the lookup happens IPv4-only (when not passing a record_type.